main.sh

```diff
@@ -29,10 +30,9 @@ detect_build_necessity() {
   esac
 
   log_info "keywords: ${keywords[*]}"
-
   for keyword in "${keywords[@]}"; do
     log_info "keyword: $keyword"
-    if grep -qw -e "$keyword" <<< "$commit_msg"; then
```

Comment:
-P is used as, `skip-` should not be a match when pattern is `skip` for example. And -P option is not available in BSD grep implemetation.